### DISCUSSION TITLE
Welcome to multidiffusion-upscaler-for-automatic1111 Discussions!

### DISCUSSION THREAD

**Discussion by pkuliyi2015** [pkuliyi2015/multidiffusion-upscaler-for-automatic1111#42] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by tsukibana** [comment]
非常厉害！

**Comment by AH225D** [comment]
Prompt control is weak.If the prompt control under txt2img 
could be as good as hires.fix, it would be perfect

**Reply by pkuliyi2015** [reply]
In fact the regional prompt control is not straightforward to use. You need to set background modifier to 0 and use your own bbox to fill the whole background. The results can like this:

![image](https://user-images.githubusercontent.com/59076257/227015997-722a3587-9a1b-4b60-b6ab-e40e8da207bb.png)

some of my friends' work:


![7V@C~27 8_Z1L(_Y~QJ0(M8](https://user-images.githubusercontent.com/59076257/227016269-88c2a840-0ab6-41c2-92a6-f0ef4dc0d6a0.jpg)

![V$9}3~IIX}MP~14%5F U`UI](https://user-images.githubusercontent.com/59076257/227016314-71a0ea6e-0a71-4c11-afd2-07e329284420.jpg)

**Reply by Boldor83** [reply]
can you give settings for img2img regional prompt? nothing works for me there

**Comment by MalfreCryvertia** [comment]
![image](https://user-images.githubusercontent.com/93036999/227316334-0fa226df-831b-4430-a2b2-b9988704b309.png)
so.. how do add these regions? im stuck at making the box and i cant find anyway using space enter rightclick 1 2 3 f1 f2 etc to make a 2nd region selection so it can be like in the example
![image](https://user-images.githubusercontent.com/93036999/227316662-d7431c0e-3044-46d9-9bbe-ec85ee323d19.png)

**Reply by Boldor83** [reply]
you have to enable first region control and then one of the regions, a small rectangle should appear. Only i still can make it do anything after that step

**Reply by MalfreCryvertia** [reply]
thank you, finally i can start experimenting with this extension that seems really wonderful just from a glance

**Reply by pkuliyi2015** [reply]
I updated the extension, so unnecessary settings all go away. Enjoy!

**Comment by ColorfanXP** [comment]
Im still kind of confused as how to use this extension. I was under the impression it was for upscaling images? But is it more for upscaling small regions of an image to get more quality?

**Reply by pkuliyi2015** [reply]
will update soon

**Reply by pkuliyi2015** [reply]
Updated and removed redundant settings. This extension is not only an upscaler now; it can directly draw images with text, put any character into any background.

**Comment by akabarin** [comment]
大佬问下开启multidiffusion后角色lora是否放在region的前景设置中，我试了下对比没有开启的出图感觉multidiffusion好像没有调用角色lora。
另外一个单独使用Tiled VAE功能时使用角色lora好像会频繁生成2个角色，我暂时还不太清楚是不是和这个功能的部分机制产生了冲突。
Hi,may I ask is the role lora placed in the foreground setting of region after multidiffusion is enabled? I tried to compare the image without multidiffusion on and it looked like multidiffusion didn't call the character lora.
In addition, when using Tiled VAE function alone, the role lora seems to generate two roles frequently. I am not sure whether there is a conflict with some mechanism of this function.

**Comment by fandui111** [comment]
![00050-2226994953-((masterpiece,best quality))](https://user-images.githubusercontent.com/60561342/229281992-e4e1b084-adba-4147-83b9-e9a3bb7da9a9.png)
启用两个插件图生图生成2k图片时，放大可发现出现了如图点状问题，1920x1080则不会，这是模型的问题还是我使用方法的问题？请教大佬该如何设置才能避免。

**Comment by Rkkss** [comment]


**Reply by LightChaser666** [reply]
Here is an example for i2i upscale. Personally I use remarry, but you can also select UltraSharp as your upscaler.
![IMG_0093](https://user-images.githubusercontent.com/103820794/229956341-675fc95c-5e1e-4884-bc07-cd0d1d8fde31.PNG)

**Reply by Rkkss** [reply]
Tried your settings, doesn't seem to improve. Would be nice if you can make some direct comparision with ultimate SD script.
I've tested various settings for two days and Ultimate is still clear winner at denoise <0.35

**Comment by WalterShomer** [comment]
Hi, any chance anybody experienced this? not sure what i'm doing wrong, read through the tutorials and posts around and it seems to be a strightforward process but somehow fail to generate a picture of a single character in higher resolution.

right now what's happening is it generates the prompt but for each tile, and than plasters all the tiles together so end result is lots of characters, tried tiled diffusion + tiled vae, high res fix + tiled vae.

does it have to be a controlnet and then break it into tiles? or only img 2 img? 

also using amd rx5700xt directml repo ( could it be because of this?)

**Reply by ColorfanXP** [reply]
Thats exactly the issue I was having and I havent been able to figure out how to resolve that. I figured that maybe I misunderstood what this was meant to do.

**Reply by PotatoBananaApple** [reply]
Most likely you have too small tile width/height or too small tile overlap. 

Also if you use Hires. fix + high denoise it can mess the image up, as it will use same size tiles for normal generate + hires.

**Reply by WalterShomer** [reply]
just to make sure 'm talking about txt2img not img2img
i'm not 100% sure but i think it eventually 'worked' with defaults by pushing decoder up which stopped the dupes until ran out of vram but there was still 1 long sim at the bottom. (thought its the same thing like someones post around here about double horizons)

also the maximum image size wasn't much bigger then already possible, ( can do 960x960 and managed to get upto 1100x1100 ).

could this potentially increase the limit to say 1.5x1.5 if setting are correct?

**Reply by PotatoBananaApple** [reply]
If you are getting duplicate things, it usually means your tile width/height or overlap is too low. If you use Hires. fix, you can't really do big upscale with high denoise without messing things up.

Doing txt2img with low hires fix and then upscaling higher in img2img is better idea than just hires fix

**Comment by Charmandrigo** [comment]
Is it lora compatible for the region prompt? 
I made a few attempts, and it's kinda strange, it only applies one lora on one region, it does not affect the other regions which is good and is exactly what I been looking for but other regions with a lora on it's prompt don't seem to apply or do anything
